rasp/selenium_engine.py

```diff
@@ -1,91 +1,31 @@
 import selenium
 import selenium.webdriver
-import time
 
 from rasp.base import Engine, Webpage
 from rasp.errors import EngineError
 
 
 class BaseSeleniumEngine(Engine):
-    def __init__(self):
-        self.driver = None
-        return
+    def __init__(self, *args, **kwargs):
+        self.driver = selenium.webdriver.Firefox()
 
     def get_source(self):
         return str(self.driver.page_source)
 
     def cleanup(self):
         self.driver.quit()
-        return
-
-    def setup(self):
-        self.driver = selenium.webdriver.Firefox()
-        return
 
     def load_page(self, url):
         self.driver.get(url)
-        return
 
     def get_url(self):
         return self.driver.current_url
 
-    def clone(self):
-        return BaseSeleniumEngine()
-
 
 class SeleniumEngine(BaseSeleniumEngine):
-    def __init__(self):
-        super(SeleniumEngine, self).__init__()
-        super(SeleniumEngine, self).setup()
-        return
-
     def get_page_source(self, url):
-        if url:
-            super(SeleniumEngine, self).load_page(url)
-            return Webpage(
-                    super(SeleniumEngine, self).get_url(),
-                    super(SeleniumEngine, self).get_source()
-                    )
-        else:
-            return None
-
-    def cleanup(self):
```

Comment:
anything in cleanup may be better suited in __del__.  In some cases it can be useful to explicitly close connections, rather than let the os clean them up, especially if you are creating/removing engines in something like a pyspark map, or you can actually run out of file handles in some cases.  So that kind of thing is good to put in a __del__, depending on how the underlying library handles closing connections.